tests/benchmarks.rs

```diff
@@ -0,0 +1,194 @@
+mod common;
+
+use std::time::Instant;
+use std::{sync::Arc, time::Duration};
+
+use bitcoin::hex::DisplayHex;
+use bitcoin::Amount;
+use common::{
+	expect_channel_ready_event, generate_blocks_and_wait, premine_and_distribute_funds,
+	setup_bitcoind_and_electrsd, setup_two_nodes_with_store, TestChainSource,
+};
+use criterion::{criterion_group, criterion_main, Criterion};
+use ldk_node::{Event, Node};
+use lightning_types::payment::{PaymentHash, PaymentPreimage};
+use rand::RngCore;
+use tokio::task::{self};
+
+use crate::common::open_channel_push_amt;
+
+fn spawn_payment(node_a: Arc<Node>, node_b: Arc<Node>, amount_msat: u64) {
+	let mut preimage_bytes = [0u8; 32];
+	rand::thread_rng().fill_bytes(&mut preimage_bytes);
+	let preimage = PaymentPreimage(preimage_bytes);
+	let payment_hash: PaymentHash = preimage.into();
+
+	// Spawn each payment as a separate async task
+	task::spawn(async move {
+		println!("{}: Starting payment", payment_hash.0.as_hex());
+
+		loop {
+			// Pre-check the HTLC slots to try to avoid the performance impact of a failed payment.
+			while node_a.list_channels()[0].next_outbound_htlc_limit_msat == 0 {
+				println!("{}: Waiting for HTLC slots to free up", payment_hash.0.as_hex());
+				tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```

Comment:
How did you determine 100ms here? Do we have an intuition if this could starve any of the waiting payers?

Reply:
It isn't great indeed. Basically something that looked 'reasonable' to me. I think it is a weakness in the API that a user even needs to do this.